### DISCUSSION TITLE
Handling authentication

### DISCUSSION THREAD

**Discussion by rtpm** [Instawork/hyperview#506] [General]
As I see there's more hype in the dev worls on sdui app development so I hope hyperscript gets more attention.

In some issue I saw you guys are  using Django as a backend for InstaWork.

Can you please share some info on how you handle authentication (especially hyperview side)  in your app or how it can be done ?

AFAIK, django session auth can't be used at the moment.

Thanks

**Comment by adamstep** [comment]
Hi @rtpm, the Instawork app started as a regular React Native app that used a JSON API. Authentication with this app was done by storing an API token on the device, and sending it with every request. Since this authentication scheme already existed, we use it for authenticating Hyperview requests as well:
- On the client, the main `Hyperview` component takes an implementation of `fetch()` used to make requests. We provide an implementation of `fetch()` that wraps the base implementation, but adds an `Authorization` request header with the API token.
- On the Django side, we added API token authentication support via [Django REST Framework](https://www.django-rest-framework.org/api-guide/authentication/#tokenauthentication).

Because Hyperview allows you to hook into `fetch()`, you could save the session cookie in async storage, and add it to requests instead of the API token. We are also open to adding full cookie support to Hyperview, our initial explorations showed that iOS/Android support was not very good, but potentially we could integrate a library like https://github.com/react-native-cookies/cookies.

**Reply by hrstoyanov** [reply]
As of 2023, the most talked-about authentication is passkeys. How can one do this with Hyperview?

https://webauthn.guide/

**Comment by hrstoyanov** [comment]
Does hyperview support  passkeys authentication?

https://developers.google.com/identity/passkeys
